src/pharmpy/tools/structsearch/tool.py

```diff
@@ -344,6 +351,32 @@ def validate_input(
                     f'Invalid dv_types key "{key}". Allowed keys are: "drug", "target" and "complex".'
                 )
 
+    if type.lower() == 'tmdd':
+        if search_space is not None:
+            raise ValueError('Search space is not defined for TMDD models.')
+        if any([b_init, emax_init, ec50_init, met_init]):
+            raise ValueError(
+                'b_init, emax_init, ec50_init and met_init are not defined for TMDD models.'
+            )
+    elif type.lower() == 'pkpd':
+        if extra_model is not None:
+            raise ValueError('Extra model is not defined for PKPD models.')
```

Comment:
To me this sounds like it should have been defined but wasn't. Perhaps `Extra model should not be defined for PKPD models.`. What do you think?

Reply:
I see what you mean. Technically, extra model is not defined for PKPD models so the error message is correct but I understand that it can be misinterpreted. Maybe just `invalid input argument extra model for PKPD models.`?

Reply:
That sounds better.